src/RegExpr.pas

```diff
@@ -4204,34 +4199,39 @@ function TRegExpr.MatchPrim(prog: PRegExprChar): boolean;
 function TRegExpr.Exec(const AInputString: RegExprString): boolean;
 begin
   InputString := AInputString;
-  Result := ExecPrim(1, False);
+  Result := ExecPrim(1, False, False);
 end; { of function TRegExpr.Exec
   -------------------------------------------------------------- }
 
 {$IFDEF OverMeth}
 function TRegExpr.Exec: boolean;
+var
+  Len: integer;
+  SlowChecks: boolean;
 begin
-  Result := ExecPrim(1, False);
+  Len := Length(fInputString);
+  SlowChecks := (Len >= fSlowChecksSizeMin) and (Len < fSlowChecksSizeMax);
```

Comment:
Why do we need min?
We do not want to use optimisation on too small (Len < fSlowChecksSizeMax) -ok.
But why DO we use optimisation on very small (for Len < fSlowChecksSizeMin SlowCheck will be OFF so we will use optimisation)?